include/libint2/dfbs_generator.h

```diff
@@ -0,0 +1,210 @@
+/*
+ *  Copyright (C) 2004-2021 Edward F. Valeev
+ *
+ *  This file is part of Libint.
+ *
+ *  Libint is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Libint is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Libint.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef _libint2_src_lib_libint_dfbs_generator_h_
+#define _libint2_src_lib_libint_dfbs_generator_h_
+
+#include <algorithm>
+#include <libint2.h>
+#include <math.h>
+#include <libint2/shell.h>
+#include <libint2/basis.h>
+#include <libint2/atom.h>
+#include <libint2/boys.h>
+#include <Eigen/Dense>
+#include <Eigen/Eigenvalues>
+#include <boost/math/special_functions/gamma.hpp>
+
+namespace libint2 {
+    typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> Matrix;
+    typedef Eigen::Matrix<int, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> Matrix_int;
+    typedef std::vector<Shell> Shellvec;
```

Comment:
do not introduce such typedefs in a namespace scope in headers unless they are part of libint2 canon. These are overly generic anyway. Use Eigen::MatrixXd and Eigen::MatrixXi directly; std::vector<Shell> is not significantly longer than Shellvec